### PR TITLE
feat: add schema to event parameters

### LINKED ISSUE #226: Python meets AppMap v1.11.0 specification for updated Parameter object types
In v1.11.0, an additional `items` property was added to the Parameter object format to describe array elements within an HTTP request or response body.
https://github.com/getappmap/appmap#parameter-object-format

If an object is array-like, use `items` to describe the type(s) of elements stored within the array.

### PR DESCRIPTION
**Pull request by apotterri**
Add "properties" and "items" schemas to "parameters" objects , as specified in v1.11.0 of the appmap spec.

Fixes #226 .

### CODE REVIEW

**Review by dustinbyrne — APPROVED** [review]
This change looks good to me.

Though in my testing, I've been unable to get a `return_value` from an `http_server_response` event using Flask 1.1.2. I'm not really sure what the deal is there. I'm not getting any exceptions so I'm inclined to say it's unrelated to this change. Here's the app if you're interested: https://gist.github.com/dustinbyrne/ef0f8094c55d5724ae6f40edbd4a3927

**Comment by apotterri** [comment]
I'll take a quick look at the Flask issue. Flask 1 got EOLed with the release of Flask 2 on 2021-05-11, though, so I'm not sure how much effort we should devote to supporting it.

File: _appmap/event.py [hunk 1]
```diff
@@ -81,7 +74,48 @@ def describe_value(val):
...
+
+    if islist:
+        elts = [(None, v) for v in val]
+        schema_key = "items"
+    elif isdict:
+        elts = val.items()
+        schema_key = "properties"
+
+    schema = [_describe_schema(k, v, depth + 1, max_depth) for k, v in elts]
+    # schema will be [None] if depth is exceeded, don't use it
```
[the remark this answers is not in the document]
Comment (apotterri) [reply _appmap/event.py:105]: I believe I ported over all the tests from `appmap-ruby`. Maybe one showing this behavior is missing?

[the remark this answers is not in the document]
Comment (apotterri) [reply _appmap/event.py:102]: I added a test to check this: https://github.com/getappmap/appmap-ruby/pull/321 .

**Review by dustinbyrne — APPROVED** [review]

**Comment by kgilpin** [comment]
> I'll take a quick look at the Flask issue. Flask 1 got EOLed with the release of Flask 2 on 2021-05-11, though, so I'm not sure how much effort we should devote to supporting it.

EOL two years ago? None :-)

**Pull request merged**